core/src/main/scala/org/bitcoins/core/gcs/BlockFilter.scala

```diff
@@ -0,0 +1,111 @@
+package org.bitcoins.core.gcs
+
+import org.bitcoins.core.crypto.DoubleSha256Digest
+import org.bitcoins.core.protocol.CompactSizeUInt
+import org.bitcoins.core.protocol.blockchain.Block
+import org.bitcoins.core.protocol.script.{EmptyScriptPubKey, ScriptPubKey}
+import org.bitcoins.core.protocol.transaction.{Transaction, TransactionOutput}
+import org.bitcoins.core.script.control.OP_RETURN
+import org.bitcoins.core.util.BitcoinSUtil
+import scodec.bits.ByteVector
+
+object BlockFilter {
+
+  /**
+    * Returns all ScriptPubKeys from a Block's outputs that are relevant
+    * to BIP 158 Basic Block Filters
+    * @see [[https://github.com/bitcoin/bips/blob/master/bip-0158.mediawiki#contents]]
+    */
+  def getOutputScriptPubKeysFromBlock(block: Block): Vector[ScriptPubKey] = {
+    val transactions: Vector[Transaction] = block.transactions.toVector
+
+    val newOutputs: Vector[TransactionOutput] = transactions.flatMap(_.outputs)
+
+    newOutputs
+      .filterNot(_.scriptPubKey.asm.contains(OP_RETURN))
+      .filterNot(_.scriptPubKey == EmptyScriptPubKey)
+      .map(_.scriptPubKey)
+  }
+
+  /**
+    * Returns all ScriptPubKeys from a Block's inputs that are relevant
+    * to BIP 158 Basic Block Filters
+    * @see [[https://github.com/bitcoin/bips/blob/master/bip-0158.mediawiki#contents]]
+    */
+  /*
+  TODO uncomment and add unit tests for this method
+  def getInputScriptPubKeysFromBlock(
```

Comment:
Is this not needed anymore? If so, let's remove?

Reply:
I don't know I commented them because they don't have test coverage to make `core` buildable.

Reply:
cc @nkohen

Reply:
If this is never used I think it is fine to delete along with `TempUtxoProvider.scala` (it was all intended to be temporary anyway)